pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -262,10 +262,29 @@ type NIC struct {
 	PXE bool `json:"pxe"`
 }
 
+// Firmware describes the firmware on the host.
+type Firmware struct {
+	// The BIOS for this firmware
+	BIOS BIOS `json:"bios"`
+}
+
+// BIOS describes the BIOS version on the host.
+type BIOS struct {
+	// The release/build date for this BIOS
+	Date string `json:"date"`
```

Comment:
Should this field be a metav1.Time instead? I don't know if we have any idea if the input format is standard enough to be able to parse it, so maybe that's why you're using a string?

Reply:
I would love to use a proper `Time` field but I believe these versions, supplied directly from the vendor during introspection will be in a variety of formats.  If there is a belief we should come up with a good way to generically try and process them I wouldn't argue.  We would have to have some empty fallback for this process because I feel quite certain we will run into a date that we cannot parse at some point.

Reply:
That's more or less what I thought. If we start seeing a few of them and it looks like maybe we could add a best-effort parser later, we can add a separate field to hold the parsed value.